File: dataset/codenet/Python/Code/s949655009.py
strn = input()
n = int(strn)
shichigolist = ["3","5","7"]
shichigo = [[] for _ in range(len(strn))]
for i in range(len(strn)):
	if i == 0:
		for j in shichigolist:
			shichigo[i].append(j)
	else:
		for j in shichigo[i-1]:
			for k in shichigolist:
				shichigo[i].append(j+k)
target = len(strn)-1
for i in range(len(strn)):
	for j in shichigo[i][:]:
		if i == target:
			if int(j) > n:
				shichigo[target].remove(j)
			elif j.count("3")==0 or j.count("5")==0 or j.count("7")==0:
				shichigo[target].remove(j)
		else:
			if j.count("3")==0 or j.count("5")==0 or j.count("7")==0:
				shichigo[i].remove(j)

print(sum([len(shichigo[i]) for i in range(len(strn))]))

File: output/deepseek_chat/codenet/direct_translation/python/java/s949655009.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class s949655009 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String strn = scanner.nextLine();
        int n = Integer.parseInt(strn);
        String[] shichigolist = {"3", "5", "7"};
        List<List<String>> shichigo = new ArrayList<>();
        for (int i = 0; i < strn.length(); i++) {
            shichigo.add(new ArrayList<>());
        }
        for (int i = 0; i < strn.length(); i++) {
            if (i == 0) {
                for (String j : shichigolist) {
                    shichigo.get(i).add(j);
                }
            } else {
                for (String j : shichigo.get(i - 1)) {
                    for (String k : shichigolist) {
                        shichigo.get(i).add(j + k);
                    }
                }
            }
        }
        int target = strn.length() - 1;
        for (int i = 0; i < strn.length(); i++) {
            for (String j : new ArrayList<>(shichigo.get(i))) {
                if (i == target) {
                    if (Integer.parseInt(j) > n) {
                        shichigo.get(target).remove(j);
                    } else if (countOccurrences(j, '3') == 0 || countOccurrences(j, '5') == 0 || countOccurrences(j, '7') == 0) {
                        shichigo.get(target).remove(j);
                    }
                } else {
                    if (countOccurrences(j, '3') == 0 || countOccurrences(j, '5') == 0 || countOccurrences(j, '7') == 0) {
                        shichigo.get(i).remove(j);
                    }
                }
            }
        }
        int sum = 0;
        for (int i = 0; i < strn.length(); i++) {
            sum += shichigo.get(i).size();
        }
        System.out.println(sum);
    }

    private static int countOccurrences(String str, char ch) {
        int count = 0;
        for (char c : str.toCharArray()) {
            if (c == ch) {
                count++;
            }
        }
        return count;
    }
}

